### PR TITLE
convert x-templates to vue SFCs

### PR DESCRIPTION
**Pull request by OmgImAlexis**
I had to turn off linting for vue files right now as most of the new ones have issues. I'll fix them and re-enable linting in another PR after this.

### CODE REVIEW

File: themes/dark/assets/js/templates/root-dirs.vue [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+            this.$emit('update:root-dirs-value', this.rootDirsValue, this.rootDirs);
+            this.$nextTick(() => {
+                // @FIXME: Temporarily trigger a regular event as well
+                $('#rootDirText').trigger('change');
+            });
+        }
+    }
+};
+</script>
+<style>
```
Comment (p0psicles) [inline themes/dark/assets/js/templates/root-dirs.vue]: Should we make sfc css scoped?

Comment (OmgImAlexis) [reply themes/dark/assets/js/templates/root-dirs.vue]: If you don't want the outside CSS to be able to override the component then yes.

**Comment by p0psicles** [comment]
EditShow broke with the branch.
Can you fix that first?

**Comment by OmgImAlexis** [comment]
@p0psicles any other pages you can see with an issue?

**Comment by OmgImAlexis** [comment]
It should be working fine now.

**Comment by OmgImAlexis** [comment]
I would suggest disabling whitespace when you review. Should bring the whole PR down to less than `100+/-`.

**Pull request merged**